src/service/url-replacements-impl.js

```diff
@@ -103,8 +115,10 @@ export class UrlReplacements {
     }));
 
     // Returns the referrer URL.
-    this.set_('DOCUMENT_REFERRER', () => {
-      return viewerFor(this.win_).getReferrerUrl();
+    this.setAsync_('DOCUMENT_REFERRER', () => {
+      return viewerFor(this.win_).getReferrerUrl().then(referrer => {
+        return referrer === undefined ? '' : referrer;
```

Comment:
No, in retrospect this should not have been necessary.  I was debating making the result explicitly string given we need to replace the macro but after some thought decided null/undefined was equivalent to empty string.  Should you decide to roll this back, I will fix.  Otherwise I can address in a subsequent PR.